ut/s3_object_api_handler_test.cc

```diff
@@ -189,18 +190,28 @@ TEST_F(S3ObjectAPIHandlerTest, ShouldCreateS3PostMultipartObjectAction) {
   S3Option::get_instance()->disable_murmurhash_oid();
 }
 
-TEST_F(S3ObjectAPIHandlerTest, DoesNotSupportCopyPart) {
+TEST_F(S3ObjectAPIHandlerTest, ShouldCreateS3PutMultiObjectCopyAction) {
   // Creation handler per test as it will be specific
+  std::map<std::string, std::string> input_headers;
+  input_headers["Authorization"] = "1";
+  EXPECT_CALL(*mock_request, get_in_headers_copy()).Times(1).WillOnce(
+      ReturnRef(input_headers));
   handler_under_test.reset(
       new S3ObjectAPIHandler(mock_request, S3OperationCode::multipart));
 
   EXPECT_CALL(*(mock_request), http_verb()).WillOnce(Return(S3HttpVerb::PUT));
   EXPECT_CALL(*(mock_request), get_header_value(StrEq("x-amz-copy-source")))
-      .WillOnce(Return("someobj"));
-
+      .WillOnce(Return("/bkt1/obj1"));
+  EXPECT_CALL(*(mock_request), get_query_string_value(_))
+      .WillRepeatedly(Return("123"));
+  EXPECT_CALL(*(mock_request), is_chunked()).WillRepeatedly(Return(true));
+  S3Option::get_instance()->enable_murmurhash_oid();
```

Comment:
Dont enable murmurhash, currently we use motr api to create oid
murmur hash algo was used previously

Reply:
removed the use of murmur hash & used motor API, but that caused the addition of motr API shared_ptr as agrs in APIHandler class constructors.